### PR TITLE
Fyle 85zrmb5uq Code Coverage : Highlight Pipe

### PR DESCRIPTION
**Pull request by CODEX108**

### CODE REVIEW

File: src/app/shared/pipes/highlight.pipe.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { HighlightPipe } from './highlight.pipe';
+
+describe('HighlightPipe', () => {
+  const highlightPipe = new HighlightPipe();
+  it('create an instance', () => {
+    expect(highlightPipe).toBeTruthy();
+  });
+
+  it('highlightPipe transform() : should highlight the search term in the text', () => {
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/shared/pipes/highlight.pipe.spec.ts:9]: replace `term` with `query` everywhere

File: src/app/shared/pipes/highlight.pipe.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    const search = 'highlight multiple';
+    const expectedResult =
+      'This is some text that we want to <span class="highlight">highlight</span> <span class="highlight">multiple</span> words';
+
+    const result = highlightPipe.transform(text, search);
+
+    expect(result).toBe(expectedResult);
+  });
+
+  it('highlightPipe transform() : should handle a search term with special character', () => {
```
Comment (Dimple16) [inline src/app/shared/pipes/highlight.pipe.spec.ts:30]: add another test case with different special characters like @#$%^

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

**Comment by CODEX108** [comment]
Can I add a single test for the special character?
```
 it('highlightPipe transform() : should handle a search query other with special characters', () => {
    const text = 'This is some text with special characters: @#$%^+-[]{}()/*';
    const search = '@#$%^+-[]{}()/*';
    const expectedResult = 'This is some text with special characters: <span class="highlight">@#$%^+-[]{}()/*</span>';

    const result = highlightPipe.transform(text, search);

    expect(result).toBe(expectedResult);
  });
```

File: src/app/shared/pipes/highlight.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const text = 'Project Dummy Test-1';
+    const search = 'Dummy Test-1';
+    const expectedResult = 'Project <span class="highlight">Dummy</span> <span class="highlight">Test-1</span>';
+
+    const result = highlightPipe.transform(text, search);
+
+    expect(result).toBe(expectedResult);
+  });
+
+  it('highlightPipe transform() : should handle a search query with special character', () => {
```
Comment (jayfyle) [inline src/app/shared/pipes/highlight.pipe.spec.ts:29]: Change spec description, should highlight search query special characters. Do ask Dimple once, as this one and below seem quite similar making it difficult to differentiate between the two specs.

File: src/app/shared/pipes/highlight.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    const text = 'Testing Project FAE 1/Test 1';
+    const search = '/';
+    const expectedResult = 'Testing Project FAE 1<span class="highlight">/</span>Test 1';
+
+    const result = highlightPipe.transform(text, search);
+
+    expect(result).toBe(expectedResult);
+  });
+
+  it('highlightPipe transform() : should handle a search query other with special characters', () => {
```
Comment (jayfyle) [inline src/app/shared/pipes/highlight.pipe.spec.ts:39]: Remove other from the spec description.

Comment (Dimple16) [reply src/app/shared/pipes/highlight.pipe.spec.ts:39]: this isn't resolved

File: src/app/shared/pipes/highlight.pipe.spec.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/shared/pipes/highlight.pipe.spec.ts:29]: rename it to `should handle a search query with single special character`

File: src/app/shared/pipes/highlight.pipe.spec.ts (hunk 4 above)
Comment (Dimple16) [reply src/app/shared/pipes/highlight.pipe.spec.ts:39]: rename it to `should handle a search query with multiple special characters`

**Review by Dimple16 — APPROVED** [review]
minor comments, fix all of them and merge

**Pull request merged**